### PR TITLE
Add PlaceId support for Geo Autocomplete feature

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
- Update existing SearchForSuggestions API to return PlaceId (if available) along with text (title, address) of the places.
- Implement a new API called SearchForLocationById which will be a wrapper around AmazonLocationService’s GetPlace API

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/geo/src/Providers/AmazonLocationServiceProvider.ts [hunk 1]
```diff
@@ -292,14 +296,59 @@ export class AmazonLocationServiceProvider implements GeoProvider {
...
+			text: result.Text,
+			placeId: result.PlaceId,
+		}));
 
 		return results;
 	}
 
+	private _verifyPlaceId(placeId: string) {
+		// if (isValid(placeId)) { - if we have more validation rules for placeId
+		if (placeId.length === 0) {
```
Comment (thaddmt) [inline packages/geo/src/Providers/AmazonLocationServiceProvider.ts:311]: We can probably check for undefined here as well. Have we asked the location team if there is a regex or something in place for the placeId? They seem to list regex for other things on their reference page - https://docs.aws.amazon.com/location/latest/APIReference/API_DescribePlaceIndex.html

File: packages/geo/src/Providers/AmazonLocationServiceProvider.ts [hunk 2]
```diff
@@ -292,14 +296,59 @@ export class AmazonLocationServiceProvider implements GeoProvider {
...
+
+		const client = new LocationClient({
+			credentials: this._config.credentials,
+			region: this._config.region,
+			customUserAgent: getAmplifyUserAgent(),
+		});
+
+		const searchForLocationByIdInput: GetPlaceCommandInput = {
+			PlaceId: placeId,
+			IndexName: options?.searchIndexName,
```
Comment (thaddmt) [inline packages/geo/src/Providers/AmazonLocationServiceProvider.ts:337]: This should default to `this._config.search_indices.default` and be options.searchIndexName if set right? Maybe something like

```
options?.searchIndexName || this._config.search_indices.default
```

**Review by thaddmt — COMMENTED** [review]
LGTM so far, just waiting on unit tests :)

File: packages/geo/src/Providers/AmazonLocationServiceProvider.ts (hunk 1 above)
Comment (sreeramsama) [reply packages/geo/src/Providers/AmazonLocationServiceProvider.ts:311]: Did not check for undefined as this function only accepts placeId as a string.
Checked with the Location Services team and they said there is no regex - and they just check for undefined or empty strings.

File: packages/geo/src/Providers/AmazonLocationServiceProvider.ts (hunk 2 above)
Comment (sreeramsama) [reply packages/geo/src/Providers/AmazonLocationServiceProvider.ts:337]: Yes. Thanks for catching that. Will update it.

File: packages/geo/__tests__/Geo.test.ts [hunk 3]
```diff
@@ -332,6 +338,45 @@ describe('Geo', () => {
 		});
 	});
 
+	describe('searchByPlaceId', () => {
+		const testPlaceId = 'a1b2c3d4';
+		const testResults = {
+			Place: TestPlacePascalCase,
+		};
+
+		test('should search with PlaceId as input', async () => {
```
Comment (thaddmt) [inline packages/geo/__tests__/Geo.test.ts:347]: Can we also test some error cases here?

File: packages/geo/__tests__/Providers/AmazonLocationServiceProvider.test.ts [hunk 4]
```diff
@@ -478,6 +484,56 @@ describe('AmazonLocationServiceProvider', () => {
 		});
 	});
 
+	describe('searchByPlaceId', () => {
+		const testPlaceId = 'a1b2c3d4';
+		const testResults = {
+			Place: TestPlacePascalCase,
+		};
+
+		test('should search with PlaceId as input', async () => {
```
Comment (thaddmt) [inline packages/geo/__tests__/Providers/AmazonLocationServiceProvider.test.ts:493]: Can we also test some error cases or invalid inputs here?

**Review by thaddmt — APPROVED** [review]

**Comment by thaddmt** [comment]
Can you ask in the JS channel if we can run integration / unit tests on a forked pr? if not maybe we should move this to a branch on amplify-js direclty so the tests run

File: packages/geo/__tests__/Geo.test.ts (hunk 3 above)
Comment (ioanabrooks) [reply packages/geo/__tests__/Geo.test.ts:347]: +1

**Review by thaddmt — APPROVED** [review]

**Pull request merged**